include/openssl/ossl_typ.h

```diff
@@ -203,6 +203,16 @@ typedef long ossl_intmax_t;
 typedef unsigned long ossl_uintmax_t;
 #endif
 
+/*
+ * Use this inside a union with the field that needs to be aligned to a
+ * reasonable boundary for the platform.  The most pessimistic alignment
+ * of the listed types will be used by the compiler.
+ */
+# define OSSL_UNION_ALIGN       \
+    double align;               \
+    ossl_uintmax_t align_int;   \
+    void *align_ptr
+
```

Comment:
I don't see the need for this to be in a public header...

Reply:
How can this be avoided when it is included by `providers/common/ciphers/ciphers_locl.h`?
I'd much prefer it to be internal (`cryptlib.h` seemed natural) but provider...

Reply:
Try this and see if your worrying too much :wink:

```
git grep internal providers/
```

Reply:
Ugh, the common provider code should only be using public interfaces shouldn't it?
How could someone else write a provider if not?

I'm happy to change this to internal though.

Reply:
How someone else writes their providers is up to them.  If they need to do the alignment game, they can probably figure out their own way to do it.

Reply:
> Ugh, the common provider code should only be using public interfaces shouldn't it?
> How could someone else write a provider if not?

This code is internal to openssl. It has no impact at all the public provider interface. We do not need to solve the problem of how external provider authors align their structures.

Reply:
Okay, a misunderstanding on my part.  I've made the switch to internal.